mobile/src/home-view.js

```diff
@@ -0,0 +1,278 @@
+import React, { Component } from 'react'
+import ReactNative, {
+  KeyboardAvoidingView, Platform, TouchableOpacity, Text, TextInput, View, ScrollView, FlatList
+} from 'react-native'
+// import { List, ListItem, SearchBar } from "react-native-elements";
+import client, { Avatar, TitleBar } from '@doubledutch/rn-client'
+import FirebaseConnector from '@doubledutch/firebase-connector'
+
+const fbc = FirebaseConnector(client, 'qaapp')
+fbc.initializeAppWithSimpleBackend()
+
+
+
+export default class HomeView extends Component {
+  constructor() {
+    super()
+    this.state = { question: '', vote: '', sharedTasks: [], sharedVotes: [], characterCount: 0, showRecent: false, newUpdate: false,}
+    this.signin = fbc.signin()
+      .then(user => this.user = user)
+      .catch(err => console.error(err))
+  }
+
+  componentDidMount() {
+    this.signin.then(() => {
+      const sharedRef = fbc.database.public.allRef('questions')
+      const sharedRefVotes = fbc.database.public.allRef('votes')
+      sharedRef.on('child_added', data => {
+        this.setState({ sharedTasks: [...this.state.sharedTasks, {...data.val(), key: data.key }] })
+      })
+      sharedRef.on('child_removed', data => {
+        this.setState({ sharedTasks: this.state.sharedTasks.filter(x => x.key !== data.key) })
+      })
+      sharedRef.on('child_changed', data => {
+        this.setState({ sharedTasks: [...this.state.sharedTasks, {...data.val(), key: data.key }] })
+      })
+      sharedRefVotes.on('child_added', data => {
+        this.setState({ sharedVotes: [...this.state.sharedVotes, {...data.val(), key: data.key }] })
+      })
+      sharedRef.on('child_removed', data => {
+        this.setState({ sharedVotes: this.state.sharedVotes.filter(x => x.key !== data.key) })
+      })
+      sharedRefVotes.on('child_changed', data => {
+        this.setState({ sharedVotes: [...this.state.sharedVotes, {...data.val(), key: data.key }] })
+      })
+    })
+  }
+
+  renderHeader = () => {
+    return <SearchBar placeholder="Type Here..." lightTheme round />;
+  };
+  renderSeparator = () => {
+    return (
+      <View
+        style={{
+          height: 1,
+          width: "86%",
+          backgroundColor: "#CED0CE",
+          marginLeft: "14%"
+        }}
+      />
+    );
+  };
+
+
+
+  render() {
+    
+    const { sharedTasks, sharedVotes, showRecent, newUpdate } = this.state
+ 
+    questions = sharedTasks.map(t => ({...t, type:'shared'}))
+    votes = sharedVotes.map(t => ({...t, type:'shared'}))
+    questionScore = questions.map(function(x) { return Object.assign({id: x.key, score: 0}) })
```

Comment:
Let's avoid doing heavy calculations in `render()` and simply display state that has already been computed.

The DB change handlers above should be responsible for updating questions and scores, and then `render()` can simply display those.  This separates the concerns of the voting/score logic from the displaying of the questions and votes, and also has performance implications, since recomputing all the scores for all the questions on every `render()` will be expensive as the number of questions grows.